clusterDeployer.py

```diff
@@ -282,22 +301,45 @@ def create_cluster(self) -> None:
         logger.info(cfg)
         self._ai.create_cluster(cluster_name, cfg)
 
-    def create_masters(self) -> None:
+    def ensure_infraenv_created(self, *, is_workers: bool) -> str:
         cluster_name = self._cc.name
-        infra_env = f"{cluster_name}-{self.masters_arch}"
+        if is_workers:
+            infra_env = f"{cluster_name}-{self.workers_arch}"
+        else:
+            infra_env = f"{cluster_name}-{self.masters_arch}"
         logger.info(f"Ensuring infraenv {infra_env} exists.")
 
         cfg = {}
         cfg["cluster"] = cluster_name
         cfg["pull_secret"] = self._secrets_path
-        cfg["cpu_architecture"] = self.masters_arch
+        if is_workers:
+            cfg["cpu_architecture"] = self.workers_arch
+        else:
+            cfg["cpu_architecture"] = self.masters_arch
         cfg["openshift_version"] = self._cc.version
         if self._cc.proxy:
             cfg["proxy"] = self._cc.proxy
         if self._cc.noproxy:
             cfg["noproxy"] = self._cc.noproxy
+
         self._ai.ensure_infraenv_created(infra_env, cfg)
 
+        return infra_env
+
+    def nodes_set_password(self, *, is_workers: bool, hosts: typing.Iterable[ClusterHost]) -> None:
```

Comment:
can we make this cleaner by importing `from typing import Iterable` at the very top instead of doing typing.Iterable since all we care about is the type.

Reply:
done.